solver/scheduler.go

```diff
@@ -351,6 +352,7 @@ func (s *scheduler) mergeTo(target, src *edge) bool {
 type edgeFactory interface {
 	getEdge(Edge) *edge
 	setEdge(Edge, *edge)
+	findMergeDirection(*edge, *edge) (*edge, *edge)
```

Comment:
Can't this logic be inside the `setEdge` instead of exposed in this interface.

Reply:
Sadly not (this would be nice though).

We need to make sure we call `s.ef.setEdge` in the right direction, but we also need to make sure we call `s.mergeTo` in the right direction as well.